lib/req/steps.ex

```diff
@@ -183,6 +183,10 @@ defmodule Req.Steps do
       iex> Req.get!("https://httpbin.org/basic-auth/foo/bar", netrc: "/path/to/custom_netrc").status
       200
 
+      iex> System.put_env("NETRC", "/path/in/environment/variable")
+      iex> Req.get!("https://httpbin.org/basic-auth/foo/bar", netrc: :env).status
```

Comment:
I think `netrc: :env` makes sense but I feel like when the user already set the environment variable we should take the hint. In other words, we add the `load_netrc` step if `netrc: true | path` or `NETRC` variable is set.

What do you think?

Reply:
I am at a disadvantage since I wasn't familiar with `.netrc` until I worked on this, so I don't know, for example, if people might have `NETRC` set but not want to use it for some reason. If it is ok to assume that if it is set is should be used, sure.

If you had `NETRC` set but didn't want to use it, and didn't want to use `.netrc` either, you would have to unset it before making this call. I don't know if that is something someone would ever want to do, or not.

Reply:
> I am at a disadvantage since I wasn't familiar with .netrc until I worked on this, so I don't know, for example, if people might have NETRC set but not want to use it for some reason

Right, I think if someone has the file and/or the env variable but they do not want them to be used it is because they want to use _different_ auth credentials. If we today set `netrc: true` and `auth: {user, pass}`, the latter will take precedence because the `auth` step runs _later_.

I had an idea in https://github.com/wojtekmach/req/issues/52#issue-1050348346 that instead of `netrc: true` we should have `auth: :netrc` and as a matter of fact make it the default. I think we should move forward with that idea then.

I think when we have `auth: :netrc`, we try to load `~/.netrc` and if it's not there, we ignore it. We can overwrite the default lookup path with `NETRC` but if we do so and the file is not there, we raise.

WDYT?

Sorry for back'n'forth!

Reply:
I would say we just make both changes at the same time as opposed t relying on the default ordering of the steps to always be preserved.  I could just close this PR and combine those two items in a new one; it's pretty minimal work.

Reply:
Yeah doing both issues in a single PR sounds good. Feel free to do it here or on a separate one, totally up to you. Thank you!